🧭 KPI 1 — Promedio de contactos necesarios para que un cliente se suscriba

⸻

🎯 Objetivo

Calcular cuántos contactos (llamadas telefónicas) de media necesita el banco para lograr que un cliente diga “sí” a una campaña de suscripción (depósito a plazo).



In [ ]:
import pandas as pd

In [14]:
# 1 - Cargar el DataFrame
df_bank = pd.read_excel("data_set_complete.xlsx")
df_bank.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 39578 entries, 0 to 39577
Data columns (total 29 columns):
 #   Column             Non-Null Count  Dtype         
---  ------             --------------  -----         
 0   age                39578 non-null  int64         
 1   job                39578 non-null  object        
 2   marital            39578 non-null  object        
 3   education          39578 non-null  object        
 4   default            39578 non-null  int64         
 5   housing            39578 non-null  int64         
 6   loan               39578 non-null  int64         
 7   contact            39578 non-null  object        
 8   duration           39578 non-null  int64         
 9   campaign           39578 non-null  int64         
 10  pdays              39578 non-null  int64         
 11  previous           39578 non-null  int64         
 12  poutcome           39578 non-null  object        
 13  emp.var.rate       39578 non-null  float64       
 14  cons.p

In [ ]:
# 2- Filtrar solo clientes que se suscribieron (y == True)
df_subs = df_bank[df_bank['y'] == True].copy()

In [16]:
# Calcular media de la columna 'campaign' para suscriptores
promedio_contactos = df_subs['campaign'].mean()

In [17]:
# Mostrar resultado
print(f"📞 Promedio de contactos para lograr una suscripción: {promedio_contactos:.2f}")

📞 Promedio de contactos para lograr una suscripción: 2.06


In [19]:
# Clientes que se suscribieron con solo 2 contacto
df_1llamada = df_subs[(df_subs['y'] == True) & (df_subs['campaign'] == 2)].copy()
total = len(df_1llamada)


# Porcentajes por condición financiera
porcentaje_prestamo = (df_1llamada['loan'] == 1).sum() / total * 100
porcentaje_hipoteca = (df_1llamada['housing'] == 1).sum() / total * 100
porcentaje_limpios = ((df_1llamada['loan'] == 0) & 
                      (df_1llamada['housing'] == 0)).sum() / total * 100
                    

# Mostrar resultados
print(f"📞 Clientes suscritos con 2 llamada: {total}")
print(f"💳 Con préstamo personal: {porcentaje_prestamo:.2f}%")
print(f"🏠 Con hipoteca: {porcentaje_hipoteca:.2f}%")
print(f"🆕 Sin productos financieros previos: {porcentaje_limpios:.2f}%")

📞 Clientes suscritos con 2 llamada: 1126
💳 Con préstamo personal: 16.25%
🏠 Con hipoteca: 55.42%
🆕 Sin productos financieros previos: 38.28%


In [20]:
# Clientes que se suscribieron con solo 1 contacto
df_1llamada = df_subs[(df_subs['y'] == True) & (df_subs['campaign'] == 1)].copy()
total = len(df_1llamada)


# Porcentajes por condición financiera
porcentaje_prestamo = (df_1llamada['loan'] == 1).sum() / total * 100
porcentaje_hipoteca = (df_1llamada['housing'] == 1).sum() / total * 100
porcentaje_limpios = ((df_1llamada['loan'] == 0) & 
                      (df_1llamada['housing'] == 0)).sum() / total * 100
                    

# Mostrar resultados
print(f"📞 Clientes suscritos con 1 llamada: {total}")
print(f"💳 Con préstamo personal: {porcentaje_prestamo:.2f}%")
print(f"🏠 Con hipoteca: {porcentaje_hipoteca:.2f}%")
print(f"🆕 Sin productos financieros previos: {porcentaje_limpios:.2f}%")

📞 Clientes suscritos con 1 llamada: 2092
💳 Con préstamo personal: 14.67%
🏠 Con hipoteca: 54.88%
🆕 Sin productos financieros previos: 39.05%


In [21]:
# Clientes que se suscribieron con 3 o + contacto
df_1llamada = df_subs[(df_subs['y'] == True) & (df_subs['campaign'] >= 3)].copy()
total = len(df_1llamada)


# Porcentajes por condición financiera
porcentaje_prestamo = (df_1llamada['loan'] == 1).sum() / total * 100
porcentaje_hipoteca = (df_1llamada['housing'] == 1).sum() / total * 100
porcentaje_limpios = ((df_1llamada['loan'] == 0) & 
                      (df_1llamada['housing'] == 0)).sum() / total * 100
                    

# Mostrar resultados
print(f"📞 Clientes suscritos con 3 o + llamada: {total}")
print(f"💳 Con préstamo personal: {porcentaje_prestamo:.2f}%")
print(f"🏠 Con hipoteca: {porcentaje_hipoteca:.2f}%")
print(f"🆕 Sin productos financieros previos: {porcentaje_limpios:.2f}%")

📞 Clientes suscritos con 3 o + llamada: 1068
💳 Con préstamo personal: 14.14%
🏠 Con hipoteca: 55.15%
🆕 Sin productos financieros previos: 39.04%


🔍 Insight KPI 1 — Promedio de contactos necesarios para lograr una suscripción

Los clientes que finalmente se suscriben requieren, en promedio, 2.06 contactos para tomar una decisión.

Este dato sugiere una alta eficacia de la campaña: con apenas un par de interacciones, el cliente medio ya responde positivamente.

La baja cantidad de contactos necesarios implica que la estrategia de comunicación es eficiente y directa, reduciendo costes y mejorando la conversión en fases tempranas del funnel.

El perfil financiero de los clientes no cambia significativamente según el número de contactos necesarios: en todos los grupos, alrededor del 55 % ya tiene hipoteca y un ~14 % préstamo personal, mientras que el ~39 % no tiene productos financieros.

Esto sugiere que la receptividad a la campaña no está fuertemente condicionada por el nivel de vinculación previa. Por tanto, incrementar el número de llamadas no modifica el perfil financiero del cliente, lo que puede cuestionar la eficiencia de campañas prolongadas.